docs/deployment/engine_on_yarn.md

```diff
@@ -117,29 +117,37 @@ Kyuubi currently does not support Spark's [YARN-specific Kerberos Configuration]
 so `spark.kerberos.keytab` and `spark.kerberos.principal` should not use now.
 
 Instead, you can schedule a periodically `kinit` process via `crontab` task on the local machine that hosts Kyuubi server or simply use [Kyuubi Kinit](settings.html#kinit).
- 
- ## Deploy Kyuubi Flink Engine on Yarn
- 
- ### Requirements
- 
- When you want to deploy Kyuubi's Flink SQL engines on YARN, you'd better have cognition upon the following things.
- 
- - Knowing the basics about [Running Flink on YARN](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/resource-providers/yarn)
- - A binary distribution of Flink which is built with YARN support
-   - Download a recent Flink distribution from the [Flink official website](https://flink.apache.org/downloads.html) and unpack it
- - An active [Apache Hadoop YARN](https://hadoop.apache.org/docs/current/hadoop-yarn/hadoop-yarn-site/YARN.html) cluster
-   - Make sure your YARN cluster is ready for accepting Flink applications by running yarn top. It should show no error messages
- - An active Apache Hadoop HDFS cluster
- - Setup Hadoop client configurations at the machine the Kyuubi server locates
- 
- ### Configurations
- 
- #### Environment
- 
- Either `HADOOP_CONF_DIR` or `YARN_CONF_DIR` is configured and points to the Hadoop client configurations directory, usually, `$HADOOP_HOME/etc/hadoop`.
- 
- If the `HADOOP_CONF_DIR` points the YARN and HDFS cluster correctly, and the `HADOOP_CLASSPATH` environment variable is set, you can launch a Flink on YARN session, and submit an example job:
- ```bash
+
+## Deploy Kyuubi Flink Engine on Yarn
```

Comment:
You only add `Yarn Session Mode` in this document. Other changes are invalid and could be rollback.

Reply:
Other changes are deleting unuseful space. Previously, there is a space at the beginning of each line